src/clarity_vm/clarity.rs

```diff
@@ -274,28 +272,33 @@ impl ClarityInstance {
     }
 
     /// Returns the Stacks epoch of the burn block that elected `stacks_block`
+    ///
+    /// If the burn height of `stacks_block` cannot be found, default to Epoch20.
     fn get_epoch_of(
         stacks_block: &StacksBlockId,
         header_db: &dyn HeadersDB,
         burn_state_db: &dyn BurnStateDB,
-    ) -> StacksEpochId {
-        match header_db.get_burn_block_height_for_block(stacks_block) {
+    ) -> StacksEpoch {
+        let epoch_found = match header_db.get_burn_block_height_for_block(stacks_block) {
             Some(burn_height) => {
                 // special case the Stacks 2.0 genesis block -- it occurs at a zero burn block height
                 if burn_height == 0 {
```

Comment:
Woah, this is absolutely wrong and broken.  The first-ever `burn_height` is going to be 666050 on mainnet, and something entirely different on testnet.  It's only 0 on regtest.  Also, why is this branch even needed?  Doesn't `get_burn_block_height_for_block()` work for all index block hashes, including the genesis one?

Reply:
This ends up being invoked a _huge_ amount in the unit tests. I think it ends up getting invoked in all of the TestPeer tests and some of the other tests as well.

I agree that it should not ever be invoked in production, so I think it should be okay to make this branch something like `cfg(test)` only.

Reply:
Yeah, at the very least this needs to be `#[cfg(test)]`.  In all non-test cases, `get_stacks_epoch()` should cover the full domain of burnchain block heights.  I.e., `get_stacks_epoch(0)` should be `Some(..)`.

Reply:
I'm trying to take out the special case, and testing it out on GitHub to see what fails on there. We can roll back.

But, seems like maybe the bandaid shouldn't go here but instead inside the mock BurnStateDB.

Reply:
I put a "cfg(test)" part in the if statement.. or if there is a different style maybe soeone can suggest it.

Reply:
Proposal is to:
1) use the "cfg(test)" fix for now
2) fix in issue #2907